packages/x-date-pickers/src/internals/hooks/useDefaultReduceAnimations.ts

```diff
@@ -0,0 +1,11 @@
+import useMediaQuery from '@mui/material/useMediaQuery';
+
+const PREFERS_REDUCED_MOTION = '@media (prefers-reduced-motion: reduce)';
+
+export const defaultReduceAnimations =
+  typeof navigator !== 'undefined' && /(android)/i.test(navigator.userAgent);
+
+export const useDefaultReduceAnimations = () => {
+  const prefersReduced = useMediaQuery(PREFERS_REDUCED_MOTION, { defaultMatches: false });
+  return prefersReduced || defaultReduceAnimations;
+};
```

Comment:
There are two different concepts here:

1. `defaultReduceAnimations` is to disable animations that are too slow on low-end devices, it's performance related
2.  `@media (prefers-reduced-motion: reduce)` is user preference related, it's motion sickness related.

I would expect 1. and 2. to not apply at the same time, since not all the animations are too slow for performance, e.g. not enough hardware acceleration, while different animations create motion sickness.

So, this logic looks strange.

Reply:
Good observation, but I'm not sure if they are that much different. Any animation has a performance impact, regardless if it's heavy or not. If we are talking about low-end devices, then it could make sense to disable all animations as well. 🤔 
But if we are on this topic, why do we consider all Android devices as low-end and disable animations on them? 🤔

Reply:
> then it could make sense to disable all animations as well.

Note that "reduced motion" doesn't mean no animations but no movement.

Reply:
https://a11y-101.com/development/reduced-motion

> On Win10 go to "Settings > Ease of Access > Display > Simplify and personalise Windows". Set "Show animation in Windows" to off.

Looks like this behavior is called "animation" on Windows

Reply:
Oh yeah, it's inconsistent https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion
<img width="739" alt="Screenshot 2023-07-31 at 16 10 54" src="https://github.com/mui/mui-x/assets/3165635/41221c72-2187-45af-9ed2-e1ad2d5bea6d">